### PR TITLE
Initial commit to add 'Run Selected Text' functionality

### PR DESCRIPTION
**Pull request by jgbradley1**
Signed-off-by: Joshua Bradley <jgbradley1@gmail.com>

PR is a proposed solution to address [this](https://github.com/theia-ide/theia/issues/4516) issue.

### CODE REVIEW

**Comment by jgbradley1** [comment]
`Run Selected Text` really functions the same way as what you would expect from a `Run Selected Line(s)` command. i.e if you select one line and part of the next line, both complete lines are sent to the terminal.

I disagree with the naming of this menu item, but VS Code started it with their implementation. I think it's more important to align the same functionality up with them for now (even if it is misnamed).

File: packages/monaco/src/browser/monaco-editor-model.ts [hunk 1]
```diff
@@ -124,8 +125,28 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
...
     }
 
-    getText(): string {
-        return this.model.getValue();
+    /**
+     * Return selected text by Range or all text by default
+     */
+    getText(range?: Range): string {
+        if (!range) { return this.model.getValue(); }
+        let text: string = '';
```
Comment (akosyakov) [inline packages/monaco/src/browser/monaco-editor-model.ts]: `return this.model.getValueInRange(this.p2m.asRange(range));`

Comment (jgbradley1) [reply packages/monaco/src/browser/monaco-editor-model.ts]: Done. That is a cleaner solution.

File: packages/task/src/browser/task-service.ts [hunk 2]
```diff
@@ -238,6 +247,24 @@ export class TaskService implements TaskConfigurationClient {
...
     }
 
+    /**
+     * Run the selected text lines in active terminal.
+     */
+    async runSelectedText(): Promise<void> {
+        if (!this.editorManager.currentEditor) { return; }
+        const startLine = this.editorManager.currentEditor.editor.selection.start.line;
+        const endLine = this.editorManager.currentEditor.editor.selection.end.line;
+        const selectedRange: Range = Range.create(startLine, 0, endLine + 1, 0);
```
Comment (akosyakov) [inline packages/task/src/browser/task-service.ts]: Why whole line? in VS Code it runs only selection:
```ts
const selectedText = this.editorManager.currentEditor.editor.document.getText(this.editorManager.currentEditor.editor.selection);
```

**Comment by akosyakov** [comment]
I'm testing with trying to run `yarn` from `doc/Developing.md`. It works in VS Code, but with this PR the whole line is executed.

File: packages/task/src/browser/task-service.ts (hunk 2 above)
Comment (jgbradley1) [reply packages/task/src/browser/task-service.ts]: My mistake. I thought VS Code behaved differently. This has been fixed.
After some testing, there are two general cases that must be covered:

1) User selects no text - VS code sends entire line that the cursor is on to the terminal
2) User selects text - VS code only sends the portion of text that is selected

I think the last commit now handles both cases.

**Pull request merged**